packages/next/next-server/server/config.ts

```diff
@@ -252,10 +259,20 @@ function assignDefaults(userConfig: { [key: string]: any }) {
           `Specified images.sizes should be an Array received ${typeof images.sizes}`
         )
       }
-      const invalid = images.sizes.filter((d: unknown) => typeof d !== 'number')
+
+      if (images.sizes.length > 50) {
+        throw new Error(
+          `Specified images.sizes exceeds length of 50, received length (${images.sizes.length}), please reduce the length of the array to continue`
+        )
+      }
+
+      const invalid = images.sizes.filter((d: unknown) => {
+        return typeof d !== 'number' || d < 0 || d > 10000
```

Comment:
Sharp requires an int greater than zero

```
Uncaught Error: Expected positive integer for width but received 0 of type number
    at Object.invalidParameterError (/Users/styfle/Code/vercel/next.js/node_modules/sharp/lib/is.js:101:10)
    at Sharp.resize (/Users/styfle/Code/vercel/next.js/node_modules/sharp/lib/resize.js:214:16)
```

```suggestion
        return typeof d !== 'number' || d < 1 || d > 10000
```